### PR TITLE
fix: match mergify automerge rules to actionlint rules

### PR DESCRIPTION
**Pull request by nathan-weinberg**
This fixes an issue I hit in https://github.com/instructlab/instructlab/pull/1724 - previously, changing a non-workflow file in `.github` would block automerge as Mergify expected a result from `actionlint`, but `actionlint` only runs when a workflow file is changed. This PR aligns the regexs we are using in the respective configurations.

**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the
  [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [ ] [Changelog](https://github.com/instructlab/instructlab/blob/main/CHANGELOG.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

File: .github/mergify.yml [hunk 1]
```diff
@@ -13,13 +13,13 @@ pull_request_rules:
...
-    # If files are changed in .github/, the actionlint check must pass
+    # If workflow configuration files in .github/ are changed, the actionlint check must pass
     - or:
       - and:
         # regex should match the one in .github/workflows/actionlint.yml
-        - files~=.github/.*$
+        - files~=.github/**/*.ya?ml$
         - check-success=actionlint
-      - -files~=.github/.*$
+      - -files~=.github/**/*.ya?ml$
```
Comment (russellb) [inline .github/mergify.yml:22]: I don't think `**` is valid here. It's not the same syntax as the github workflows. (Of course ...)

Here we have a Python style regex.  Something like ... `\.github/.*\.ya?ml`

I would have to use the mergify config editor and validate it against some PRs to make sure it's correct, though

Comment (nathan-weinberg) [reply .github/mergify.yml:22]: Ah regex my old friend

Comment (nathan-weinberg) [reply .github/mergify.yml:22]: That config is valid according to the UI dashboard

File: .github/workflows/actionlint.yml [hunk 2]
```diff
@@ -7,14 +7,14 @@ on:
       - "main"
       - "release-**"
     paths:
-      - '.github/workflows/*.ya?ml'
+      - '.github/**/*.ya?ml'
```
Comment (bjhargrave) [inline .github/workflows/actionlint.yml:10]: Not sure why this change. Workflows are always in .github/workflows. So widening this for actionlint is wrong.

**Review by bjhargrave — CHANGES_REQUESTED** [review]

File: .github/workflows/actionlint.yml (hunk 2 above)
Comment (bjhargrave) [reply .github/workflows/actionlint.yml:10]: And this is not a regex. :-)

Comment (nathan-weinberg) [reply .github/workflows/actionlint.yml:10]: `actions/action.yml`?

Comment (bjhargrave) [reply .github/workflows/actionlint.yml:10]: `actions` holds a composite workflow (this is new since I last looked :-). So that could be `.github/actions/**/*.ya?ml` then.

Comment (nathan-weinberg) [reply .github/workflows/actionlint.yml:10]: Ideally, that action could live in the `workflows` dir... @russellb can I move it there? or does it need to be under an `actions/` directory?

Comment (bjhargrave) [reply .github/workflows/actionlint.yml:10]: I think it is common convention to put composite actions under `.github/actions` but I don't believe it is required.

Comment (russellb) [reply .github/workflows/actionlint.yml:10]: i think i followed docs to put it there

and I thought the changed looked appropriate so it found changes under actions/ as well

**Comment by RobotSail** [comment]
I marked this as part of milestone for 0.19.0, but feel free to correct me if I am wrong.

**Comment by nathan-weinberg** [comment]
@RobotSail all good, this one is def not critical to the project :+1:

File: .github/workflows/actionlint.yml (hunk 2 above)
Comment (nathan-weinberg) [reply .github/workflows/actionlint.yml:10]: I went with what I hope is a happy medium here

**Review by russellb — APPROVED** [review]

File: .github/mergify.yml [hunk 3]
```diff
@@ -13,13 +13,13 @@ pull_request_rules:
     - label!=needs-rebase
     - check-success=DCO
 
-    # If files are changed in .github/, the actionlint check must pass
+    # If workflow configuration files in .github/ are changed, the actionlint check must pass
     - or:
       - and:
         # regex should match the one in .github/workflows/actionlint.yml
-        - files~=.github/.*$
+        - files~=\.github/.*\.ya?ml
```
Comment (bjhargrave) [inline .github/mergify.yml:20]: Per the comment immediately above, the regex should match the globs in actionlint.yml. So

```regex
\.github/(actions|workflows)/(.*\.ya?ml|actionlint\..*)

```

**Review by bjhargrave — CHANGES_REQUESTED** [review]

File: .github/mergify.yml (hunk 3 above)
Comment (nathan-weinberg) [reply .github/mergify.yml]: Done

**Review by bjhargrave — APPROVED** [review]
